src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -49,6 +53,37 @@ internal AnalysisResultBuilder(bool logAnalyzerExecutionTime, ImmutableArray<Dia
             return map;
         }
 
+        private static ImmutableDictionary<string, OneOrMany<AdditionalText>> CreatePathToAdditionalTextMap(ImmutableArray<AdditionalText> nonSourceFiles)
+        {
+            if (nonSourceFiles.IsEmpty)
+            {
+                return ImmutableDictionary<string, OneOrMany<AdditionalText>>.Empty;
+            }
+
+            var builder = ImmutableDictionary.CreateBuilder<string, OneOrMany<AdditionalText>>(PathUtilities.Comparer);
+            foreach (var file in nonSourceFiles)
+            {
+                // Null file path for additional files is not possible from IDE or command line compiler host.
+                // However, it is possible from custom third party analysis hosts.
+                // Ensure we handle it gracefully
+                var path = file.Path ?? string.Empty;
+
+                // Handle multiple additional files with same path.
```

Comment:
>Handle multiple additional files with same path. [](start = 19, length = 48)

What is the scenario for multiple additional files with the same path?

Reply:
It is allowed by the command line compiler. I have added a unit test with multiple additional files with same path in this PR, which was crashing here without handling that case.